src/sortStudents.ts

```diff
@@ -1,16 +1,102 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'avg',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
   // write your function
+  const clonedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        clonedStudents.sort(
+          (student, studen2) => student[sortBy].localeCompare(studen2[sortBy]),
+        );
+        /* Please excuse the variable name studen2, had issues with
+        max length and typescript seems to not like line breaks in
+        arrow functions (the linter kept yelling at me) */
+        break;
```

Comment:
Have you tried something like this?
```suggestion
        clonedStudents.sort(
          (student, secondStudent) => student[sortBy]
            .localeCompare(studen2[sortBy]),
        );
        break;
```

Reply:
Looks wild to me, I will give it a try